web-ng/src/app/services/project/project.service.ts

```diff
@@ -137,5 +137,25 @@ export class ProjectService {
       .toList()
       .sortBy(entry => entry.email)
       .toArray();
+   }
+  /** 
+   * Checks if a user can add points to a specific layer.
+   */
+  canUserAddPointsToLayer(layer: Layer): boolean {
+    const user = this.authService.getCurrentUser();
+    if (!user) {
+      return false;
+    }
+    const userRole = this.currentProject.acl.get(user.email);
+    switch (userRole) {
```

Comment:
i think this block can be simplified as something like:
```ts
return isManager(project) || (isContributor(project) && layer.contributorsCanAdd?.includes('points'))

isManager(user) {
 ...
return  [Role.OWNER, Role.MANAGER].includes(role);
}
```

 etc.

Reply:
updated.